nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/main/java/org/apache/nifi/excel/ExcelRecordReader.java

```diff
@@ -45,8 +46,14 @@ public class ExcelRecordReader implements RecordReader {
     private final String dateFormat;
     private final String timeFormat;
     private final String timestampFormat;
+    private final DataFormatter dataFormatter;
 
     public ExcelRecordReader(ExcelRecordReaderConfiguration configuration, InputStream inputStream, ComponentLog logger) throws MalformedRecordException {
+        this(configuration, inputStream, logger, null);
+    }
+
+    public ExcelRecordReader(ExcelRecordReaderConfiguration configuration, InputStream inputStream, ComponentLog logger, Locale locale) throws MalformedRecordException {
```

Comment:
Instead of adding the new constructor, recommend changing all references to the single constructor with the `Locale` required, defaulting to `Locale.getDefault()`